File: solutions/349-intersection-of-two-arrays/intersection-of-two-arrays.ts
function intersection(nums1: number[], nums2: number[]): number[] {

    const set1 = new Set<number>(nums1);
    const res: number[] = [];

    for(let i = 0; i < nums2.length; i += 1) {
        if(set1.has(nums2[i])) {
            res.push(nums2[i]);
            set1.delete(nums2[i]);
        }
    }
    
    return res;
};

File: solutions/349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        
        res = []
        set_1 = set(nums1)
        set_2 = set(nums2)

        for num in set_1:
            if num in set_2:
                res.append(num)

        return res